xonsh/imphooks.py

```diff
@@ -0,0 +1,78 @@
+"""Import hooks for importing xonsh source files. This module registers
+the hooks it defines when it is imported.
+"""
+from __future__ import print_function, unicode_literals
+import os
+import sys
+import builtins
+from importlib.machinery import ModuleSpec
+from importlib.abc import MetaPathFinder, SourceLoader
+
+from xonsh.tools import string_types
+from xonsh.execer import Execer
+
+class XonshImportHook(MetaPathFinder, SourceLoader):
+    """Implements the import hook for xonsh source files."""
+
+    def __init__(self, *args, **kwargs):
+        super(XonshImportHook, self).__init__(*args, **kwargs)
+        self._filenames = {}
+        self._execer = None
+
+    @property
+    def execer(self):
+        if hasattr(builtins, '__xonsh_execer__'):
+            execer = builtins.__xonsh_execer__
+            if self._execer is not None:
+                self._execer = None
+        elif self._execer is None:
+            self._execer = execer = Execer(unload=False)
+        else:
+            execer = self._execer
+        return execer
+    #
+    # MetaPathFinder methods
+    #
+    def find_spec(self, fullname, path, target=None):
+        """Finds the spec for a xonsh module if it exists."""
+        spec = None
+        path = sys.path if path is None else path
```

Comment:
So I think that the way to do what you want is through the traditional Python module loading hook.  Namely:

```python
import sys
sys.path.insert(0, '.')
```

I don't think that this is the sort of decision that xonsh should make for users since it could also affect Python module loading.  It is better for folks who want this behaviour to stick it in their xonshrc, I think.